packages/yew/src/html/component/scope.rs

```diff
@@ -271,40 +312,25 @@ impl<COMP: BaseComponent> Scope<COMP> {
     }
 
     /// Send a message to the component.
-    ///
-    /// Please be aware that currently this method synchronously
-    /// schedules a call to the [Component](crate::html::Component) interface.
     pub fn send_message<T>(&self, msg: T)
     where
         T: Into<COMP::Message>,
     {
-        self.push_update(UpdateEvent::Message(msg.into()));
+        // We are the first message in queue, so we queue the update.
+        if self.pending_messages.push(msg.into()) == 1 {
+            self.push_update(UpdateEvent::Message);
+        }
     }
 
     /// Send a batch of messages to the component.
-    ///
-    /// This is useful for reducing re-renders of the components
-    /// because the messages are handled together and the view
-    /// function is called only once if needed.
-    ///
-    /// Please be aware that currently this method synchronously
-    /// schedules calls to the [Component](crate::html::Component) interface.
     pub fn send_message_batch(&self, messages: Vec<COMP::Message>) {
-        // There is no reason to schedule empty batches.
-        // This check is especially handy for the batch_callback method.
-        if messages.is_empty() {
-            return;
+        for message in messages {
+            self.send_message(message);
```

Comment:
Could `extend` the `pending_messages` instead of iterating, that'd bump the capacity of the vec at most once.